restapi/auth.go

```diff
@@ -33,47 +34,29 @@ func generateStateOauthCookie(w http.ResponseWriter) (string, error) {
 	return state, nil
 }
 
-func (api api) Login(w http.ResponseWriter, r *http.Request) render.Renderer {
-	state, err := generateStateOauthCookie(w)
-	if err != nil {
-		return rest.ErrInternal(api.logger, err)
-	}
-
-	url := api.oauthconfig.AuthCodeURL(state)
-	http.Redirect(w, r, url, http.StatusTemporaryRedirect)
-
-	return rest.JSONStatusOK(url)
+type GoogleClaims struct {
+	Email string `mapstructure:"email"`
+	Hd    string `mapstructure:"hd"`
 }
 
-func (api api) AuthCallback(w http.ResponseWriter, r *http.Request) render.Renderer {
-	expectedState, _ := r.Cookie("oauthstate")
-	if expectedState.Value != r.FormValue("state") {
-		return rest.ErrUnauthorized("Invalid oauth state")
-	}
+func (api api) Login(w http.ResponseWriter, r *http.Request) render.Renderer {
 
-	// exchange received authorization code for an access token
-	token, err := api.oauthconfig.Exchange(context.Background(), r.FormValue("code"))
+	payload, err := idtoken.Validate(oauth2.NoContext, r.Header.Get("Authorization"), config.GetGoogleClientId())
```

Comment:
what's `idtoken`?

Reply:
So the frontend will esentially handle the actual google sign in and when its successful google will return a token. This token is sent to the backend so we can verify and add the user to our db. idtoken is a library that is a wrapper around the google api that allows us to decode the token the frontend sends us